Content.Server/Communications/CommunicationsConsoleSystem.cs

```diff
@@ -0,0 +1,225 @@
+using System.Globalization;
+using Content.Server.Access.Systems;
+using Content.Server.AlertLevel;
+using Content.Server.Chat.Managers;
+using Content.Server.Popups;
+using Content.Server.RoundEnd;
+using Content.Server.Station.Systems;
+using Content.Shared.Access.Components;
+using Content.Shared.Access.Systems;
+using Content.Shared.Communications;
+using Robust.Shared.Player;
+
+namespace Content.Server.Communications
+{
+    public sealed class CommunicationsConsoleSystem : EntitySystem
+    {
+        [Dependency] private readonly IChatManager _chatManager = default!;
+        [Dependency] private readonly RoundEndSystem _roundEndSystem = default!;
+        [Dependency] private readonly AlertLevelSystem _alertLevelSystem = default!;
+        [Dependency] private readonly StationSystem _stationSystem = default!;
+        [Dependency] private readonly IdCardSystem _idCardSystem = default!;
+        [Dependency] private readonly AccessReaderSystem _accessReaderSystem = default!;
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+
+        private const int MaxMessageLength = 256;
+
+        public override void Initialize()
+        {
+            // All events that refresh the BUI
+            SubscribeLocalEvent<AlertLevelChangedEvent>(OnAlertLevelChanged);
+            SubscribeLocalEvent<CommunicationsConsoleComponent, ComponentInit>((_, comp, _) => UpdateBoundUserInterface(comp));
+            SubscribeLocalEvent<RoundEndSystemChangedEvent>(_ => OnGenericBroadcastEvent());
+            SubscribeLocalEvent<AlertLevelDelayFinishedEvent>(_ => OnGenericBroadcastEvent());
+
+            // Messages from the BUI
+            SubscribeLocalEvent<CommunicationsConsoleComponent, CommunicationsConsoleSelectAlertLevelMessage>(OnSelectAlertLevelMessage);
+            SubscribeLocalEvent<CommunicationsConsoleComponent, CommunicationsConsoleAnnounceMessage>(OnAnnounceMessage);
+            SubscribeLocalEvent<CommunicationsConsoleComponent, CommunicationsConsoleCallEmergencyShuttleMessage>(OnCallShuttleMessage);
+            SubscribeLocalEvent<CommunicationsConsoleComponent, CommunicationsConsoleRecallEmergencyShuttleMessage>(OnRecallShuttleMessage);
+        }
+
+        public override void Update(float frameTime)
+        {
+            foreach (var comp in EntityManager.EntityQuery<CommunicationsConsoleComponent>())
+            {
+                if (comp.AnnouncementCooldownRemaining <= 0f)
+                {
+                    // TODO: Find a less ass way of refreshing the UI
+                    if (comp.AlreadyRefreshed) continue;
+                    UpdateBoundUserInterface(comp);
+                    comp.AlreadyRefreshed = true;
+                    continue;
+                }
+                comp.AnnouncementCooldownRemaining -= frameTime;
+            }
+
+            base.Update(frameTime);
+        }
```

Comment:
Shouldn't you just check AlreadyRefreshed first? Then you don't need to touch any accumulation.